antispam/caches/redis/redis.py

```diff
@@ -20,13 +20,168 @@
 FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 DEALINGS IN THE SOFTWARE.
 """
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, List, AsyncIterable
+
+from attr import asdict
+
+import orjson as json
+
 from antispam.abc import Cache
+from antispam.exceptions import GuildNotFound, MemberNotFound
+from antispam.dataclasses import Message, Member, Guild, Options
+from antispam.enums import ResetType
+
+if TYPE_CHECKING:
+    from redis import asyncio as aioredis
+
+    from antispam import AntiSpamHandler
+
+log = logging.getLogger(__name__)
 
 
 class RedisCache(Cache):
-    """Not implemented lol"""
+    """
+    A cache backend built to use Redis.
+
+    Parameters
+    ----------
+    handler: AntiSpamHandler
+        The AntiSpamHandler instance
+    redis: redis.asyncio.Redis
+        Your redis connection instance.
+    """
+
+    def __init__(self, handler: AntiSpamHandler, redis: aioredis.Redis):
+        self.redis: aioredis.Redis = redis
+        self.handler: AntiSpamHandler = handler
+
+    async def get_guild(self, guild_id: int) -> Guild:
+        log.debug("Attempting to return cached Guild(id=%s)", guild_id)
+        resp = await self.redis.get(f"GUILD:{guild_id}")
+        if not resp:
+            raise GuildNotFound
+
+        resp = resp.decode("utf-8")
+        as_json = json.loads(resp)
+
+        guild: Guild = Guild(**as_json)
+        # This is actually a dict here
```

Comment:
Some form of table showing the stored data structure as it is in redis would be nice for future work. Think example dict of types

Reply:
N/A